src/components/structures/MatrixChat.js

```diff
@@ -1258,6 +1268,22 @@ export default React.createClass({
         this._setPageSubtitle();
     },
 
+    /**
+     * Called when the session is softly logged out
+     */
+    _onSoftLogout: function() {
+        this.notifyNewScreen('soft_logout');
+        this.setStateForNewView({
+            view: VIEWS.SOFT_LOGOUT,
+            ready: false,
+            collapseLhs: false,
+            collapsedRhs: false,
+            currentRoomId: null,
+            page_type: PageTypes.RoomDirectory,
```

Comment:
Not sure if `RoomDirectory` is the intended page type here, but if it is, I'd maybe put a comment here explaining why.

Reply:
Copy/paste from above. I'm not sure what would be correct, or if it's even needed...